packages/kit/src/runtime/server/cookie.js

```diff
@@ -20,23 +16,25 @@ const decode = decodeURIComponent;
  */
 export function get_cookies(request, url, dev, trailing_slash) {
 	const header = request.headers.get('cookie') ?? '';
-	const initial_cookies = parse(header, { decode });
+	const initial_cookies = parse(header, { decode: (value) => value });
 
 	const normalized_url = normalize_path(url.pathname, trailing_slash);
 	// Emulate browser-behavior: if the cookie is set at '/foo/bar', its path is '/foo'
 	const default_path = normalized_url.split('/').slice(0, -1).join('/') || '/';
 
 	if (dev) {
+		// TODO this could theoretically be wrong if the cookie was set unencoded?
```

Comment:
I'm not sure it makes sense to add a todo because I don't know how we'd ever fix it. Maybe change the comment like yhis
```suggestion
		// This could be wrong if the cookie was set unencoded
		// but since it's just for printing a debug message we can live with that
```

Reply:
I have some nebulous thoughts how this could be done, for example through never decoding it and then updating that object with an unencoded version of the new cookies, but want to look into that later - so I'd like to keep the TODO for now (but you may very well be right!).